### ISSUE TITLE
Cumulative sum

### ISSUE THREAD

**Issue by tonyxiao** [data-forge/data-forge-ts#45]
Panda has `cumsum` for its series. What's the equivalent in data forge? It doesn't seem like it exists upon first look, what's the recommended workaround?

**Comment by ashleydavis** [comment]
Data-Forge has a sum function for series that sums the values in the series. 

What does the cumsum function do?

**Comment by tonyxiao** [comment]
It turns `[1,2,3,4]` into `[1,3,6,10]`.

**Comment by tonyxiao** [comment]
So summing the values up cumulatively

**Comment by ashleydavis** [comment]
Ok, no there's nothing like that built in. Wouldn't be too hard to write a function to do that though. Are you able to point me to a blog post online (or some other docs) that shows how to use this in Pandas?

**Comment by tonyxiao** [comment]
I actually don't actually know how to use this in Panda (never used panda before) just stumbled into that function while looking for it in dataforge. I was looking for it in data-forge and couldn't find it so ended up writing this myself

```ts

      let balance = 0
      const summaryByPeriod = deltaByPeriod.toArray().map(r => {
        const startingBalance = balance
        balance += r.quantity.sum

        return {
          periodStart: r.periodStart,
          startingBalance,
          endingBalance: balance,
        }
      })

      return new Series(summaryByPeriod)
```

**Comment by ashleydavis** [comment]
Cool, that works!

If you do happen to see any examples of how this is used in Pandas please let me know. I can easily implement functionality from Pandas, but it's best if I can see an example of it first.

**Comment by zyzhu** [comment]
See pandas doc here
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.cumsum.html

**Comment by ashleydavis** [comment]
Thanks @zyzhu! Does anyone want to implement a cumsum function for Data-Forge?

**Comment by ashleydavis** [comment]
The cumsum function has been implemented for Series and published in Data-Forge at version 1.8.8.

**Comment by tonyxiao** [comment]
just discovered it in latest docs. thank you!
